src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java

```diff
@@ -370,28 +370,36 @@ void distributeFlushTasks() {
 
         if (!channelsDataPerTable.isEmpty()) {
           int idx = 0;
+          float totalBufferSizePerTableInBytes = 0F;
           while (idx < channelsDataPerTable.size()) {
             ChannelData<T> channelData = channelsDataPerTable.get(idx);
             // Stop processing the rest of channels when needed
             if (idx > 0
                 && shouldStopProcessing(
```

Comment:
I approved by accident, sorry. I think this PR still doesn't protect us from generating too large files due to asynchronous flush. There is no guarantee that between this check and actual flush, there aren't additional rows added to the buffer, right? Didn't we want to throw an exception at flush time when we see a large blob?

Reply:
I believe you can reset your status. We first create a snapshot so this size won't change anymore, all additional rows will be added to a new buffer